src/cpprealm/persisted.hpp

```diff
@@ -75,8 +85,9 @@ struct persisted_base {
     template <type_info::TimestampPersistable X, typename U, typename V>
     friend persisted<X>& operator +=(persisted<X>& a, std::chrono::duration<U, V> b);
     type as_core_type() const;
-    void assign(const Obj& object, const ColKey& col_key);
+    void assign(const Obj& object, const ColKey& col_key, SharedRealm);
```

Comment:
Why was `SharedRealm` added? We should probably start adding docstrings for the internal methods.

Reply:
We need a `SharedRealm` to construct an object-store `List`

Reply:
What I'll do is overload `assign` to only take `SharedRealm` in the `persisted_container_base` case